lib/companies_web/controllers/admin/pending_change_controller.ex

```diff
@@ -20,14 +20,14 @@ defmodule CompaniesWeb.Admin.PendingChangeController do
       {:ok, _approved_changes} ->
         conn
         |> put_flash(:info, "Changes submitted")
-        |> redirect(to: Routes.pending_change_path(conn, :index))
+        |> redirect(to: Routes.pending_change_path(conn, :index, conn.params["locale"]))
```

Comment:
If possible I would like to try and refactor out the `conn.params["locale"]` calls that come with this PR. I think there are two ways we can accomplish this:

1. Pattern matching in the function head

```elixir
def update(conn, %{"id" => change_id, "approval" => approval, "locale" => locale}) do
  case PendingChanges.approve(change_id, approval == "true") do
    {:ok, _approved_changes} ->
      conn
      |> put_flash(:info, "Changes submitted")
      |> redirect(to: Routes.pending_change_path(conn, :index, locale))

    {:error, reason} ->
      Logger.error(reason)

      conn
      |> put_flash(:error, "Changes failed")
      |> redirect(to: Routes.pending_change_path(conn, :index, locale))
  end
end
```

2. With a helper function (`UserHelpers` perhaps?):

```elixir
def update(conn, %{"id" => change_id, "approval" => approval}) do
  case PendingChanges.approve(change_id, approval == "true") do
    {:ok, _approved_changes} ->
      conn
      |> put_flash(:info, "Changes submitted")
      |> redirect(to: Routes.pending_change_path(conn, :index, locale(conn)))

    {:error, reason} ->
      Logger.error(reason)

      conn
      |> put_flash(:error, "Changes failed")
      |> redirect(to: Routes.pending_change_path(conn, :index, locale(conn)))
  end
end


defmodule CompaniesWeb.UserHelpers do
  def locale(%{params: %{"locale" => locale}}), do: locale
  def locale(_), do: "en"
end
```

A helper is probably preferably as we can use it in our views to remove the same `@conn.params["locale"]` code there.

Reply:
We could also access the `:locale` assign in place of `params`, directly, which would rely on `SetLocale` as we do elsewhere: https://github.com/smeevil/set_locale/blob/master/lib/set_locale.ex#L37

Reply:
I use this solution.